experimental/origin-dapp2/src/pages/growth/mutations/Enroll.js

```diff
@@ -0,0 +1,132 @@
+            return <QueryError error={error} query={profileQuery} />
+          }
+
+          const accountId = data.web3.primaryAccount.id
+
+          return (
+            <Mutation
+              mutation={GrowthEnroll}
+              onCompleted={({ enroll }) => {
+                if (enroll.error === null && enroll.authToken) {
+                  localStorage.setItem('growth_auth_token', enroll.authToken)
+                  this.props.history.push('/campaigns')
+                } else {
+                  console.log('Error occurred: ', enroll)
+                  this.setState({
+                    error: 'Can not enroll into growht campaign'
+                  })
+                }
+              }}
+              onError={errorData => {
+                console.log('Error: ', errorData)
+                this.setState({
+                  error: 'Problems enrolling into growth campaign.'
+                })
+              }}
+            >
+              {enroll => (
+                <Mutation
+                  mutation={SignMessageMutation}
+                  onCompleted={({ signMessage }) => {
+                    console.log('Message successfuly signed: ', signMessage)
+                    enroll({
+                      variables: {
+                        accountId: accountId,
+                        agreementMessage: this.state.message,
+                        signature: signMessage
+                      }
+                    })
+                  }}
+                  onError={errorData =>
+                    this.setState({
+                      error: 'Message signature unsuccessful',
+                      errorData
+                    })
+                  }
+                >
+                  {signMessage => (
+                    <Fragment>
+                      <div className="growth-enrollment">
+                        <video
+                          className="metamask-video"
+                          width="320"
+                          heigh="240"
+                          onLoadStart={() => {
+                            signMessage({
+                              variables: {
+                                address: accountId,
+                                // TODO: change version programatically
+                                message:
+                                  'I accept the terms of growth campaign version: 1.0'
+                              }
+                            })
+                          }}
+                          autoPlay
+                          loop
+                        >
+                          <source
+                            src="images/growth/metamask_in_browser_dark_bg.mp4"
+                            type="video/mp4"
+                          />
+                          Your browser does not support the video tag.
+                        </video>
+                        <div className="title">Confirm Metamask Signature</div>
```

Comment:
Since the wallet provider can vary, probably need to take out MetaMask references. Maybe we can just add a todo for now

Reply:
👍 added a todo